dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/association/AbstractOrganisationUnitAssociationsQueryBuilder.java

```diff
@@ -0,0 +1,216 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.association;
+
+import static java.util.stream.Collectors.joining;
+import static org.hisp.dhis.hibernate.jsonb.type.JsonbFunctions.CHECK_USER_ACCESS;
+import static org.hisp.dhis.hibernate.jsonb.type.JsonbFunctions.CHECK_USER_GROUPS_ACCESS;
+import static org.hisp.dhis.hibernate.jsonb.type.JsonbFunctions.EXTRACT_PATH_TEXT;
+import static org.hisp.dhis.hibernate.jsonb.type.JsonbFunctions.HAS_USER_GROUP_IDS;
+import static org.hisp.dhis.hibernate.jsonb.type.JsonbFunctions.HAS_USER_ID;
+import static org.hisp.dhis.security.acl.AclService.LIKE_READ_METADATA;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import lombok.RequiredArgsConstructor;
+
+import org.hisp.dhis.commons.collection.CollectionUtils;
+import org.hisp.dhis.user.CurrentUserGroupInfo;
+import org.hisp.dhis.user.CurrentUserService;
+import org.hisp.dhis.user.User;
+
+@RequiredArgsConstructor
+public abstract class AbstractOrganisationUnitAssociationsQueryBuilder
+{
+
+    private final CurrentUserService currentUserService;
+
+    private static final String SHARING_OUTER_QUERY_BEGIN = "select " +
+        "    inner_query_alias.uid, " +
+        "    inner_query_alias.agg_ou_uid " +
+        "from (";
+
+    private static final String SHARING_OUTER_QUERY_END = ") as inner_query_alias";
+
+    private static final String RELATIONSHIP_TABLE_ALIAS = "relationship_table_alias";
+
+    private static final String BASE_TABLE_ALIAS = "base_table_alias";
+
+    private static final String INNER_QUERY_GROUPING_BY = "group by " + BASE_TABLE_ALIAS + ".uid, " + BASE_TABLE_ALIAS
+        + ".sharing";
+
+    private String getInnerQuerySql()
+    {
+        return "select " +
+            "    " + BASE_TABLE_ALIAS + ".uid, " +
```

Comment:
I think the SQL here can be formatted to be more readable. I know the auto-formatter might kick in.

Reply:
yes I'll try to format it better and exclude from auto-formatting, thank you